test/var.cpp

```diff
@@ -125,40 +127,51 @@ TYPED_TEST(Var, DimCPPSmall)
 
     readTests<TypeParam, outType, double> (TEST_DIR"/var/var.data",numDims,in,tests);
 
-    for(size_t i = 0; i < in.size(); i++)
-    {
-        array input(numDims[i], &in[i].front(), afHost);
-
-        array bout  = var(input, false);
-        array nbout = var(input, true);
-
-        array bout1  = var(input, false, 1);
-        array nbout1 = var(input, true,  1);
-
-        vector<vector<outType> > h_out(4);
-
-        h_out[0].resize(bout.elements());
-        h_out[1].resize(nbout.elements());
-        h_out[2].resize(bout1.elements());
-        h_out[3].resize(nbout1.elements());
-
-        bout.host(  &h_out[0].front());
-        nbout.host( &h_out[1].front());
-        bout1.host( &h_out[2].front());
-        nbout1.host(&h_out[3].front());
-
-        for(size_t j = 0; j < tests.size(); j++) {
-            for(size_t jj = 0; jj < tests[j].size(); jj++) {
-                // NOTE: will work for all types
-                if (is_same_type<float, outType>::value ||
-                    is_same_type<cfloat, outType>::value) {
-                    ASSERT_FLOAT_EQ(real(h_out[j][jj]), real(tests[j][jj]));
-                    ASSERT_FLOAT_EQ(imag(h_out[j][jj]), imag(tests[j][jj]));
-                } else {
-                    ASSERT_DOUBLE_EQ(real(h_out[j][jj]), real(tests[j][jj]));
-                    ASSERT_DOUBLE_EQ(imag(h_out[j][jj]), imag(tests[j][jj]));
-                }
+    array input(numDims[testNum], &in[testNum].front(), afHost);
```

Comment:
Couple of things, This is not related to the fix, is it ?

And I don't think it is doing what you are expecting it to do. `readTests` function is to be called once, but moving the for loop associated with number of tests to the caller(above `testCppVarDim`) level, you are reading the test data many times over where one read should be sufficient enough.

Reply:
There is only one test so this shouldn't be an issue.
Also, we shouldn't be writing large test cases that handle multiple things. We need to be able to figure out what is going wrong just by looking at the tests that pass and tests that fail. If we can't do that then we waste time debugging tests. I tried to fix this test but it required more effort than I was willing to spend on this. I will come back to it at a later time.

Reply:
That may be the case, but that is not the point I was trying to make. `readTests` is being used incorrectly and anyone who looks at this file for reference will be mislead as to how that function works and it can lead to whole new set of problems. This change shouldn't go in this PR and should be made separately whenever we will fix this later.

Reply:
Addressed